### PR TITLE
Forks n clones

### PR DESCRIPTION
**Pull request by Tim-Jackins**
## Proposed Changes

  - Added a lesson on forking and cloning.
    - I think I explained fork and cloning accurately but I want to double check. I also think I may have been too wordy.

### CODE REVIEW

**Comment by Tim-Jackins** [comment]
>The plethora of grammatical errors throughout the entire lesson...

Wow, I should really just run stuff through grammarly or something before pushing.

>find a student to test these lessons on

Agreed! So far that student has been Shields but I'm going to try and run it by Zoe, Tilden and if I can some youngins in band.

File: engineering/3_forks_and_clones.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Forks and Clones
+
+Now you know how to make a repo, attach a remote, and put your work on GitHub! As impressive as this is it's only half the fun. The other half is collaborating with others online.
+
+"Online collaboration with classmates? Ugh, that means we have to pass around my laptop and have everyone slowly type their emails into a drive folder... Right?"
```
[the remark this answers is not in the document]
Comment (Tim-Jackins) [reply engineering/3_forks_and_clones.md]: I'll raise the criticisms of "how slow it would be to load drive on a pi" or "having to copy code to other computers before upload".

[the remark this answers is not in the document]
Comment (Tim-Jackins) [reply engineering/3_forks_and_clones.md]: Now that I'm thinking about it I'm going to move the google drive dissing into Bring a Remote Into the Mix section of the [2nd lesson](https://github.com/chssigma/git-lessons/blob/master/engineering/2_hello_github.md/)

[the remark this answers is not in the document]
Comment (Tim-Jackins) [reply engineering/3_forks_and_clones.md]: On second I don't think I will. How do you feel about that change

[the remark this answers is not in the document]
Comment (jonahweissman) [reply engineering/3_forks_and_clones.md]: I could see it either way; your choice

File: engineering/3_forks_and_clones.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+While this is the norm for services like drive, Git has a different way of going at it.
+
+Let's look at [cool-repo](https://github.com/chssigma/cool-repo). Look through it's contents to get a sense of what you need to add. I'll give you a hint it's your GitHub info and name to a particular table. So now you know what you need to add... but can you? You can't commit changes to this repo, because you haven't been added as a collaborator, but there is an interesting work around.
+
+It's time to fork and clone!
+
+Fork
```
[the remark this answers is not in the document]
Comment (Tim-Jackins) [reply engineering/3_forks_and_clones.md]: Sadly I can't keep the exclamation mark and make it a header. :disappointed:

[the remark this answers is not in the document]
Comment (jonahweissman) [reply engineering/3_forks_and_clones.md]: Nah, I meant the word "Fork"

[the remark this answers is not in the document]
Comment (jonahweissman) [reply engineering/3_forks_and_clones.md]: It looks weird by itself on a line like that

[the remark this answers is not in the document]
Comment (Tim-Jackins) [reply engineering/3_forks_and_clones.md]: I see what your saying

File: engineering/3_forks_and_clones.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+Forking is when you make an editable copy a repo to a different location on GitHub's servers.
+
+Click on the "Fork" button in the top right corner of the webpage. Once the animation is finished you should see your new forked repo!
```
[the remark this answers is not in the document]
Comment (Tim-Jackins) [reply engineering/3_forks_and_clones.md]: I think they'll have `cool-repo` open already? I added a link anyways.

File: engineering/3_forks_and_clones.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+git clone https://github.com/<your GitHub Username>/<your repos name>.git
+```
+
+You now have your very own copy of cool-repo! add your name to the appropriate table and commit and push your changes.
+
+You're a majority of the way done with the tutorial! Here's a cool diagram describing what you did:
+
+![clone](/images/clone.png)
+
+So now you need a way to get the changes you just commited back into the main repository owned by chssigma. Lucky for you Git has a way to do this built in, it's called a pull request.
```
[the remark this answers is not in the document]
Comment (Tim-Jackins) [reply engineering/3_forks_and_clones.md]: I'll refer to it as a GitHub feature

**Comment by Tim-Jackins** [comment]
Added fixes in the above 3 commits :arrow_double_up:

File: engineering/3_forks_and_clones.md [hunk 5]
```diff
@@ -1,45 +1,39 @@
...
-Let's look at [cool-repo](https://github.com/chssigma/cool-repo). Look through it's contents to get a sense of what you need to add. I'll give you a hint it's your GitHub info and name to a particular table. So now you know what you need to add... but can you? You can't commit changes to this repo, because you haven't been added as a collaborator, but there is an interesting work around.
+Let's look at [cool-repo](https://github.com/chssigma/cool-repo). Look through its contents to get a sense of what you need to add. I'll give you a hint: it's your GitHub info and name to a particular table. So now you know what you need to add... but can you? You can't push commits to this repo, because you haven't been added as a collaborator, but that won't stop you from collaborating!
 
 It's time to fork and clone!
 
-Fork
+## Fork
 
-Forking is when you make an editable copy a repo to a different location on GitHub's servers.
+Forking is when you copy a repo to a different location on GitHub's servers. Because you have ownership over this copy, you can push commits to it.
```
Comment (jonahweissman) [inline engineering/3_forks_and_clones.md]: Yeah, I like this better

File: engineering/2_hello_github.md [hunk 6]
```diff
@@ -23,12 +23,18 @@ git diff
...
 What's more fun than a repo stored in one place? A repo stored in two places! In order to share your code with teammates, you'll need to put it somewhere *remote*ly where you can both reach it. GitHub provides this service for free! Once your repo is on GitHub, other people, can pull/(copy and download) changes from the remote and (if you let them) push/(copy and upload) their changes to the remote.
 
 ![remotes](../images/remotes.png)
 
+This is the structure Git uses for online collaboration.
+
+"Did someone say online collaboration? Easy peasy, I'll make a drive folder and share it with everyone but wait... How am I going to get code from my pi onto google drive?"
+
+While this is the norm for services like drive, Git has a better way of going about it.
+
```
Comment (jonahweissman) [inline engineering/2_hello_github.md]: This feels really out of place here

File: engineering/3_forks_and_clones.md [hunk 7]
```diff
@@ -1,45 +1,39 @@
...
 
 ```bash
-git clone https://github.com/<your GitHub Username>/<your repos name>.git
+git clone https://github.com/<your GitHub Username>/cool-repo.git
 ```
 
-You now have your very own copy of cool-repo! add your name to the appropriate table and commit and push your changes.
-
-You're a majority of the way done with the tutorial! Here's a cool diagram describing what you did:
+You now have your very own local copy of your fork of `cool-repo`! Add your name to the appropriate table and commit and push your changes. Here's a cool diagram describing what you did:
```
Comment (jonahweissman) [inline engineering/3_forks_and_clones.md]: this is pretty wordy. Maybe scrap the "your very own" and just say "a"

File: engineering/3_forks_and_clones.md (hunk 5 above)
Comment (Tim-Jackins) [reply engineering/3_forks_and_clones.md]: It's definitely better for actually explaining what you're doing.

File: engineering/2_hello_github.md (hunk 6 above)
Comment (Tim-Jackins) [reply engineering/2_hello_github.md]: Hmm what about this:

"Did someone say online collaboration? I already know how to collaborate online, I'll make a drive folder and share it with everyone but wait... How am I going to get code from my pi onto google drive?"

While drive works well for sharing in some cases, Git has a better way of going about it especially relating to code.

Comment (jonahweissman) [reply engineering/2_hello_github.md]: I think that's good. Just make sure to capitalize Google Drive and Pi, and comma before "especially".

File: engineering/3_forks_and_clones.md (hunk 7 above)
Comment (Tim-Jackins) [reply engineering/3_forks_and_clones.md]: [77cfe01](https://github.com/chssigma/git-lessons/pull/10/commits/77cfe0130ad1c158bdc2270b29b5397cd31a4f1f)
